City-of-Helsinki/helsinki-paatokset
# PP-199: Case and decision pages

This includes a bunch of changes for case and decision handling, including templates that should match the previous issue and agenda_item layouts.

**To test**
* Checkout branch
* Run `make new` (Take a database dump first to store your previous state)
* SSH into the container and run `drush ap:fs;drush mim ahjo_cases:latest;drush mim ahjo_decisions:latest;drush mim ahjo_meetings:latest`
* Check out the decisions list: https://helsinki-paatokset.docker.so/fi/admin/content/decisions
  * All of the decisions should be missing the "Issued date" information
  * Open some decisions and confirm that the "Decision Record" fields are empty
* To update the decisions with missing information, either:
  * If you have VPN access, set up the Open ID settings and run `drush ap:ud -v`
  * If you don't have VPN access, run `drush ap:ud --localdata -v`
  * (If you migrated all four thousand decisions, you might want to limit the previous command with the `--limit=100` parameter.)
* Reload the decisions list. The issued date should now be filled on all of them (unless you used the limit parameter).
  * Reload the decisions you opened previously. The decision record fields should be filled in.
  * Check that the "Issued" field in the JSON matches with the Issued Date field (same timezone, etc).
  * If you used the `--localdata` parameter, some decisions might be missing the record. In this case the date should be set to "2001-01-01".
* Open some cases, for example:
  * https://helsinki-paatokset.docker.so/fi/asia/hel-2020-006250
  * https://helsinki-paatokset.docker.so/fi/asia/hel-2021-013311
  * https://helsinki-paatokset.docker.so/fi/asia/hel-2022-000167
  * The decision content is always just "Content.", but make sure the navigation links and dropdown work, and that the titles, labels, attachments and the decision PDF button's link update accordingly.
